aten/src/ATen/record_function.h

```diff
@@ -262,8 +295,11 @@ struct TORCH_API RecordFunction {
     StringView name_;
     int64_t sequence_nr_ = -1;
     std::vector<c10::IValue> inputs_;
+    std::vector<c10::IValue> outputs_;
```

Comment:
if they're not always present, then it might make sense to explicitly model that using a `c10::optional`

Reply:
The vector is empty is a meaningful check for inputs or outputs. If a callback needs inputs or outputs, it expect 0 or more data. Using `c10::optional`introduces a new possible outcome (`c10::optional::has_value()`) to check and the meaning of `optional::has_value() -> false` and `vector::empty()` seem redundant.

Reply:
The current implementation represents two things in the same way
  - needsOutput==true with an side-effect-only op returning zero returns
  - needsOutput==false with any op
Explicit representations with `c10::optional` are usually better and have a lower chance of somebody writing code that uses the `outputs_` variable forgetting that an empty vector can mean two different things.

Reply:
Yes, good points. Here it is being consistent with `inputs_` and existing interface.